loky/backend/resource_tracker.py

```diff
@@ -183,32 +205,39 @@ def main(fd, verbose=0):
             pass
 
     if verbose:  # pragma: no cover
-        sys.stderr.write("Main semaphore tracker is running\n")
+        sys.stderr.write("Main resource tracker is running\n")
         sys.stderr.flush()
 
-    cache = set()
+    cache = {rtype: set() for rtype in _CLEANUP_FUNCS.keys()}
     try:
-        # keep track of registered/unregistered semaphores
-        with os.fdopen(fd, 'rb') as f:
+        # keep track of registered/unregistered resources
+        if sys.platform == "win32":
+            fd = msvcrt.open_osfhandle(fd, os.O_RDONLY)
+        with open(fd, 'rb') as f:
             for line in f:
                 try:
-                    cmd, name = line.strip().split(b':')
-                    if cmd == b'REGISTER':
-                        name = name.decode('ascii')
-                        cache.add(name)
+                    cmd, rtype, name = line.strip().decode('ascii').split(
+                        ':', 2)
+                    cleanup_func = _CLEANUP_FUNCS.get(rtype, None)
+                    if cleanup_func is None:
```

Comment:
The way it is written here is how we ended up writting in in python/cpython#13222.

Reply:
Yes they are equivalent but it is harder to read so I think it would be better to change it. (when I read the code, I look where the `cleanup_func` variable is used)